### PR TITLE
[FLINK-14306][python] Add the code-generated flink_fn_execution_pb2.py to the codebase

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request fixes the flink-python build fails of "No module named pkg_resources" by adding the code-generated flink_fn_execution_pb2.py to the Flink codebase.*

## Brief change log

  - *Adds flink_fn_execution_pb2.py to Flink codebase*
  - *Removes the build process of flink_fn_execution_pb2.py from the pom file of flink-python*
  - *Adds tests test_flink_fn_execution_pb2_synced to make sure that flink_fn_execution_pb2.py is synced with flink_fn_execution.proto*

## Verifying this change

This change added tests and can be verified as follows:

  - *Added tests test_flink_fn_execution_pb2_synced*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by dianfu** [comment]
@hequn8128 Thanks for your review. Have addressed your comments in the latest PR. Looking forward to your feedback.

File: flink-python/pyflink/gen_protos.py [hunk 1]
```diff
@@ -137,10 +137,51 @@ def _install_grpcio_tools_and_generate_proto_files():
...
+                '#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n'
+                '#  See the License for the specific language governing permissions and\n'
+                '# limitations under the License.\n'
+                '################################################################################\n'
+            )
+            tmp_file.write(original_data)
+            os.rename(os.path.join(dir, tmp_file_name), os.path.join(dir, file_name))
+
+
+def _check_grpcio_tools_version():
```
Comment (hequn8128) [inline flink-python/pyflink/gen_protos.py]: Call this method after the `grpcio-tools` has been installed? i.e., call it after line 90. In this way, the method can be simplified a bit.

File: flink-python/pyflink/gen_protos.py [hunk 2]
```diff
@@ -137,10 +137,51 @@ def _install_grpcio_tools_and_generate_proto_files():
...
+    from pkg_resources import parse_version
+    try:
+        version = parse_version(pkg_resources.get_distribution("grpcio-tools").version)
```
Comment (hequn8128) [inline flink-python/pyflink/gen_protos.py]: change to `version = pkg_resources.get_distribution("grpcio-tools").parsed_version`

File: flink-python/pyflink/gen_protos.py (hunk 1 above)
Comment (dianfu) [reply flink-python/pyflink/gen_protos.py]: There are two reasons blocking us to do that:
1. Modules dynamically added to "sys.path" is not possible to get via `pkg_resources.get_distribution`
2. If `grpcio-tools` is not present in the Python environment and is installed via this script, we can make sure that the version is correct.

Does that make sense?

Comment (hequn8128) [reply flink-python/pyflink/gen_protos.py]: We can use `pkg_resources.working_set.add_entry` to update entrySet dynamically.

**Comment by hequn8128** [comment]
@dianfu Thanks a lot for the update. Will merge once travis passed.

**Pull request closed without merging**